addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/connector/WebInterface.java

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.internal.connector;
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.net.CookieStore;
+import java.net.HttpCookie;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpStatus.Code;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.solaredge.config.SolarEdgeConfiguration;
+import org.openhab.binding.solaredge.handler.SolarEdgeHandler;
+import org.openhab.binding.solaredge.internal.command.PostLoginGetClientCookie;
+import org.openhab.binding.solaredge.internal.command.PostLoginGetSpringSecurityToken;
+import org.openhab.binding.solaredge.internal.command.PreLogin;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The connector is responsible for communication with the solaredge webportal
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class WebInterface {
+
+    private static final long LOGIN_WAIT_TIME = 1000;
+
+    private final Logger logger = LoggerFactory.getLogger(WebInterface.class);
+
+    /**
+     * Configuration of the bridge from
+     * {@link org.openhab.BoxHandler.fritzaha.handler.FritzAhaBridgeHandler}
+     */
+    private final SolarEdgeConfiguration config;
+
+    /**
+     * Bridge thing handler for updating thing status
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * holds authentication status
+     */
+    private boolean authenticated = false;
+
+    /**
+     * holds cookie status
+     */
+    private boolean cookieOK = false;
+
+    /**
+     * HTTP client for asynchronous calls
+     */
+    private HttpClient asyncclient;
+    /**
+     * Maximum number of simultaneous asynchronous connections
+     */
+    private int asyncmaxconns = 20;
+
+    /**
+     * Constructor to set up interface
+     *
+     * @param config Bridge configuration
+     */
+    public WebInterface(SolarEdgeConfiguration config, SolarEdgeHandler handler) {
+        this.config = config;
+        this.handler = handler;
+        asyncclient = new HttpClient(new SslContextFactory(true));
+        asyncclient.setMaxConnectionsPerDestination(asyncmaxconns);
+        try {
+            asyncclient.start();
+        } catch (Exception e) {
+            logger.error("Could not start HTTP Client");
```

Comment:
Do not log to error. It's used for when openHAB fails. This is probably an info log. Also don't catch Exception, but only specific exceptions.

Reply:
If "Exception" is thrown I must catch it, there is no way out here.